rfc/text/advanced/ap_48_uris.md

```diff
@@ -2,6 +2,62 @@
 
 WAMP pre-defines the following error URIs for the **Advanced Profile**. WAMP peers SHOULD only use the defined error messages.
 
+## Authentication
+
+No authentication method the *Client* offered is accepted.
+
+{align="left"}
+        wamp.error.no_auth_method
+
+The *Client* authenticated for a *Realm* (`realm|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_realm
+
+The *Client* authenticated for a *Role* (`authrole|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_role
+
+The *Client* authenticated for a *Principal* (`authid|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_principal
+
+The authentication as presented by the *Client* is denied (e.g. "wrong password").
+
+{align="left"}
+        wamp.error.authentication_denied
+
+The authentication of the *Client* failed technically at run-time, and the *Client* is therefor rejected ("fail secure operation").
+
+{align="left"}
+        wamp.error.authentication_failed
+
+The *Client* did not authenticate (at all) and a successful (non-anonymous) authentication is required.
+
+{align="left"}
+        wamp.error.authentication_required
+
+## Authorization
+
+The *Principal* is not authorized to perform such *Action*.
```

Comment:
Here is an introduced kind of new term *Principal*. AFAIK we do not use it across the SPEC (can't find anywhere in the spec). Imo it is a pretty wide term. Why not keep the *Client*?

Btw, [Authentication Methods](https://wamp-proto.org/wamp_latest_ietf.html#name-authentication-methods) uses Client everywhere. So I think it is important to keep this all in sync. Because in this case reader is already familiar with the terms he/she saw in the spec and nothing surprising here. If we want to add new terms - they should be added and updated across all the relevant sections of the spec and also given some descriptions.

Reply:
rgd "Principal", that's right!

IMO, it is another gap. The historic reason is: we initially said in the spec "all authentication and authorization is implementation specific (outside the spec)", then we added authentication methods, but did not spec any authorization mechanisms.

- **A**uthentication: who is it?
- **A**uthorization: is it (the action by "who" on the respective "object") allowed?
- **A**uditability: what was actually done in the past (a log)

now that we want to have authorization related URIs in the spec, we a term for "who"

A WAMP *Session* is established originating from a *Client* to a *Router*, and when the *Session* performs *Authentication*, and that authentication is successful, the *Router* will have established

* *Realm*
* *AuthRole*
* *AuthId*

_for_ that *Session* running _in_ that *Client*.

The triple "(Realm, AuthRole, AuthId)" is a *Principal*.

-----

> Btw, [Authentication Methods](https://wamp-proto.org/wamp_latest_ietf.html#name-authentication-methods) uses Client everywhere

this section is about Authorization, not Authentication. this is the 2nd A in "AAA"

Reply:
I am inclined to remove the section https://github.com/wamp-proto/wamp-proto/blob/master/rfc/text/basic/bp_01_introduction.md#router-implementation-specifics completely.

what's not in the spec is not in the spec and obviously then implementation defined. why state a conclusive list when we're working on such things to add to the spec. it doesn't add much value IMO

Reply:
Yeah, I see.

My suggestion is to: 

1. Leave the term `Сlient` for this PR and merge it. It will be consistent with other chapters. This will push us a step forward.
2. Continue AAA refactoring in a separate issue.

Reply:
> I am inclined to remove the section https://github.com/wamp-proto/wamp-proto/blob/master/rfc/text/basic/bp_01_introduction.md#router-implementation-specifics completely.

Yeah 👍🏻  Why even try to mention things that lies out of the SPEC scope at all :)